src/com/zebrunner/jenkins/pipeline/tools/scm/bitbucket/BitBucket.groovy

```diff
@@ -0,0 +1,52 @@
+package com.zebrunner.jenkins.pipeline.tools.scm.bitbucket
+
+import com.zebrunner.jenkins.pipeline.tools.scm.Scm
+import com.zebrunner.jenkins.pipeline.Configuration
+
+class BitBucket extends Scm {
+
+    BitBucket(context, host, org, repo, branch) {
+        super(context, host, org, repo, branch)
+        this.prRefSpec = '+refs/pull/*:refs/remotes/origin/pr/*'
+        this.branchSpec = "%s"
+    }
+
+    BitBucket(context) {
+        super(context)
+    }
+
+    enum HookArgs {
```

Comment:
only bitbucket related?

Reply:
There is a specific branch for each scm that we are going to support(at least the pr decoration feature). From the generc webhook perspective all scm are the same if they are using the same api and share the branchSpec and prRefSpec for cloning a PR.